src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.cs

```diff
@@ -248,9 +248,8 @@ internal void SetItem(string propertyName, JsonNode? value)
 
             OrderedDictionary<string, JsonNode?> dict = Dictionary;
 
-            if (!dict.TryAdd(propertyName, value))
+            if (!dict.TryAdd(propertyName, value, out int index))
```

Comment:
I'm getting the CI build error locally as well. I think the issue is that .net 9.0 doesn't have the new `TryAdd(..., out int index)` method (the failure indicates that the S.T.J. build for 9.0 is causing the error). Should I be `ifdef`ing here to check for that?

Reply:
I think ifdefing is fine. The alternative would be to polyfill the current version of OrderedDictionary for `net9.0` as we do with the older targets, but I don't think the change is worth the duplication.

Reply:
Not sure about the style we use for `#if`s but I wanted to avoid having the `#if` region cross into the if block.